tikv-client/src/main/java/com/pingcap/tikv/meta/Collation.java

```diff
@@ -264,10 +266,113 @@ public static int translate(String collation) {
   }
 
   public static String translate(int code) {
+    if (code < 0 && isNewCollationEnabled()) {
+      code = -code;
+    }
     String collation = collationCodeMap.get(code);
     if (collation == null) {
       return "";
     }
     return collation;
   }
+
+  public static void setNewCollationEnabled(boolean flag) {
+    newCollationEnabled = flag;
+  }
+
+  public static Boolean isNewCollationEnabled() {
+    return newCollationEnabled;
+  }
+
+  // RewriteNewCollationIDIfNeeded rewrites a collation id if the new collations are enabled.
+  // When new collations are enabled, we turn the collation id to negative so that other the
+  // components of the cluster(for example, TiKV) is able to aware of it without any change to
+  // the protocol definition.
+  // When new collations are not enabled, collation id remains the same.
+  public static int rewriteNewCollationIDIfNeeded(int id) {
```

Comment:
why not just alter the getCollationCode with the same logic

Reply:
@shiyuhang0 Thanks for the review. 
getCollationCode has different meanings with rewriteNewCollationIDIfNeeded. getCollationCode return the code of collation, while rewriteNewCollationIDIfNeeded return new code for new collation. We only need to rewrite the code when we send a message to tikv or other components. getCollationCode will be used in any other place to get the code.

Reply:
@shiyuhang0 Thanks for the review.
getCollationCode has different meanings with rewriteNewCollationIDIfNeeded. getCollationCode return the code of collation, while rewriteNewCollationIDIfNeeded return new code for new collation. We only need to rewrite the code when we send a message to tikv or other components. getCollationCode will be used in any other place to get the code.